In [ ]:
! python main_binary.py --model vgg_cifar10_binary

for this version, Datatype must be float tensor,modified would come as soon as possible
saving to ./results/2019-10-30_09-36-51
creating model vgg_cifar10_binary
created model with configuration: {'input_size': None, 'dataset': 'cifar10'}
number of parameters: 36461450
Files already downloaded and verified
Files already downloaded and verified
training regime: {0: {'optimizer': 'Adam', 'betas': (0.9, 0.999), 'lr': 0.005}, 40: {'lr': 0.001}, 80: {'lr': 0.0005}, 100: {'lr': 0.0001}, 120: {'lr': 5e-05}, 140: {'lr': 1e-05}}
/usr/local/anaconda3/lib/python3.7/site-packages/torch/nn/modules/container.py:92: UserWarning: Implicit dimension choice for log_softmax has been deprecated. Change the call to include dim=X as an argument.
  input = module(input)

 Epoch: 1	Training Loss 1.8327 	Training Prec@1 34.002 	Training Prec@5 84.108 	Validation Loss 1.6560 	Validation Prec@1 39.980 	Validation Prec@5 90.390 


 Epoch: 2	Training Loss 1.4593 	Training Prec@1 50.528 	Training Prec@5 92.218 	Val

- ResNet

17528 paramers 

Epoch55 91.438%
- Binary ResNet

175406 paramerter 10 times more paramerters!

epoch2500 67.816%  can't converge

- Vgg

36461450
epoch 41 89.620% 	

- Our model

6367370 89.005%